lib/Makefile.am

```
@@ -151,10 +158,16 @@ checksrc:
 if NOT_CURL_CI
 if DEBUGBUILD
 # for debug builds, we scan the sources on all regular make invokes
-all-local: checksrc
+CHECKSOURCES = checksrc
 endif
 endif
 
+all-local: $(CHECKSOURCES) $(UNITPROTOS)
+
+# UNITPROTOS depends on every C file in the lib/ dir
+$(UNITPROTOS): $(CSOURCES)
+	@PERL@ $(top_srcdir)/scripts/extract-unit-protos $(CSOURCES) > $(UNITPROTOS)
```

```suggestion
	(cd $(srcdir) && @PERL@ $(top_srcdir)/scripts/extract-unit-protos $(CSOURCES) > $(top_builddir)/lib/$(UNITPROTOS))
```
Perhaps this for out-of-tree builds? (I haven't tested it)